### PR TITLE
Switch to go1.20 in CI

### PR DESCRIPTION
**Pull request by donatello**
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: .github/workflows/go-cross.yml [hunk 1]
```diff
@@ -20,11 +20,11 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        go-version: [1.19.x]
+        go-version: [1.20]
```
Comment (harshavardhana) [inline .github/workflows/go-cross.yml]: 1.20.x

File: .github/workflows/go-fips.yml [hunk 2]
```diff
@@ -20,17 +20,17 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        go-version: [1.19.x]
+        go-version: [1.20]
```
Comment (harshavardhana) [inline .github/workflows/go-fips.yml]: 1.20.x

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by donatello** [comment]
@klauspost @harshavardhana We have a bunch of failures under Windows in Go 1.20

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#4c975f6b-e242-44b4-baa9-3c26b818f03d) |
| mint-pools.sh | :x: [more...](#51cb3144-cfd9-4013-ac12-3c18360f081f) |
<a name="51cb3144-cfd9-4013-ac12-3c18360f081f"></a>
### 16743-3cb79b2/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.5:30363
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 7dae7a0cf051:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 7 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 8 seconds
(5/14) Running awscli tests ... done in 1 minutes and 16 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 18 seconds
(8/14) Running minio-go tests ... FAILED in 1 minutes and 24 seconds
{
  "args": {
    "bucketName": "minio-go-test-l6kgdbabojfkv9oj",
    "region": "eu-west-1"
  },
  "duration": 83655,
  "error": "Your previous request to create the named bucket succeeded and you already own it.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... done in 37 seconds
(9/14) Running minio-js tests ... done in 53 seconds
(10/14) Running minio-py tests ... done in 4 minutes and 11 seconds
(11/14) Running s3cmd tests ... done in 17 seconds
(12/14) Running s3select tests ... done in 5 seconds
(13/14) Running versioning tests ... done in 3 minutes and 6 seconds

Executed 13 out of 14 tests successfully.
```
<a name="4c975f6b-e242-44b4-baa9-3c26b818f03d"></a>
### 16743-3cb79b2/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.5:32709
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp bf82dcbcccd3:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 9 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 9 seconds
(5/14) Running awscli tests ... done in 1 minutes and 16 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... FAILED in 26 seconds
{
  "name": "mc",
  "duration": "11122",
  "function": "test_find_empty",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-9495 --quiet --no-color rb --force myminio/mc-test-bucket-19450  >>> \nmc: <ERROR> Unable to validate target `myminio/mc-test-bucket-19450`. Get \"http://15.15.15.5:32709/mc-test-bucket-19450/?location=\": dial tcp 15.15.15.5:32709: i/o timeout"
}
(7/14) Running minio-go tests ... done in 2 minutes and 0 seconds
(8/14) Running minio-java tests ... done in 30 seconds
(9/14) Running minio-js tests ... done in 51 seconds
(10/14) Running minio-py tests ... done in 3 minutes and 56 seconds
(11/14) Running s3cmd tests ... done in 16 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**